lib/katello/tasks/upgrades/3.14/republish_deb_metadata.rake

```diff
@@ -0,0 +1,24 @@
+namespace :katello do
+  namespace :upgrades do
+    namespace '3.14' do
+      desc "Regenerate the metadata for repositories affected by pulp_deb migration 0004"
+      task :republish_deb_metadata => ["environment"] do
+        User.current = User.anonymous_api_admin
+        input_file_name = "/var/lib/pulp/0004_deb_repo_republish_candidates.txt"
```

Comment:
I would strongly recommend to avoid relying on a shared filesystem. This will break on split deployments. Those are currently rare, but they are high on my wishlist. It will also make Katello much more container friendly.

Reply:
I will keep that in mind for the future, but I don't think it is worth the effort to change this for the current case. The existing db migration on the pulp side is pretty much set in stone, (though it might be possible to parse the same information from the existing logs as well).

This rake task is essentially a convenience improvement for a presumably very small number of Katello deb content users. If the `/var/lib/pulp/0004_deb_repo_republish_candidates.txt` file is missing on the disk the task just won't do anything (which is desirable for all the Katello users who don't have any deb content).

Everything fixed by this task can also be achieved by manual (though cumbersome) action. The main thing that won't work without running the rake task, is synchronizing deb content to smart proxy (which has never worked in the past).

Reply:
Can you help me understand what's in this file? Is it all Debian repositories or a subset? Would be ok to query the Pulp server for all Deb repos and process them all?

Reply:
Sure, I feel like I should have anticipated this question and answered it preemptively.

The file contains a list of pulp_ids (one per line) of every repository that had to be migrated by pulp_deb migration 0004.

This will be any Debian or Ubuntu repositories affected by a certain bug in pulp_deb. Known examples include all releases from the `debian-security` repository, as well as any Ubuntu releases that have a `-` in their name (e.g.: `bionic-backports`, `bionic-updates`, etc.)

Reply:
Since "republishing metadata" for deb type repositories, takes some amount of time (bearable but not super efficient), I would not recommend republishing all deb type repositories instead. (Though that would also solve the problem).

Reply:
So it's not all repositories but a subset. It sounds like this can be ok, but I would make sure it also works when the file exists but is not readable (because of permissions for example).

Another option to consider is making the path configurable (with its current value as the default). This would allow users of a split deployment to copy the file and manually run the task. https://www.viget.com/articles/protip-passing-parameters-to-your-rake-tasks/ has some examples.

Reply:
I replaced `File.file?` with `File.readable?`. I may have another look at the parameters later.

Reply:
I now added the ability to manually rerun the rake task using a different input file (supplied via a parameter). I also adjusted the output to reflect this.

Can I resolve this discussion?

Reply:
Yes, this addresses my concerns. Thanks!